wgsl/index.bs

```diff
@@ -2954,6 +2954,13 @@ The variable's [=store type=] must be a [=atomic-free=] [=plain type=].
 A variable or constant declared in the first clause of a `for` statement is available for use in the second
 and third clauses and in the body of the `for` statement.
 
+An instance of a function scope variable is a [=dynamic context=]. Each
+variable that is [=in scope=] for some invocation has a unique [=memory view=].
+When a variable goes out of scope, the [=memory location|memory locations=]
```

Comment:
A "unique memory view" may end up being too weak. It's the storage that doesn't overlap.  In #1735, access mode is added as another intrinsic part of the memory view, but works against the intent here.

It should apply across all variables, not just function scope (because resources never alias in WebGPU).

There's a rule in "var and let" that was meant to cover the distinctness aspect:

> Two variables with overlapping lifetimes will not have overlapping storage.

Unfortunately nothing defines the "lifetime".  But maybe the right thing to do is define lifetime somewhere, and rely on that general statement.

Reply:
I took a stab at defining lifetime and using that. PTAL